ocs_ci/ocs/resources/pvc.py

```diff
@@ -162,6 +163,27 @@ def resize_pvc(self, new_size, verify=False):
             )
         return True
 
+    def get_attached_pods(self):
+        """
+        Get the pods attached to the PVC represented by this object instance
+
+        Returns:
+            list: A list of pod objects attached to the PVC
+
+        """
+        # Importing from pod inside, because of unsolvable import loop
+        from ocs_ci.ocs.resources.pod import get_all_pods, get_pvc_name
+        attached_pods = []
+        all_pods = get_all_pods()
+        for pod_obj in all_pods:
+            try:
+                pvc = get_pvc_name(pod_obj)
+            except UnavailableResourceException:
```

Comment:
Saw the conversation with @ebenahar.
I'm not sure why raising exception is the way we prefer. 
This is not the approach in all resources and not aligning (iiuc) with this function usecase - to get the pvc attached to a pod.
In most cases we are just returning whatever exists in the object or trying to align with oc commands. In this case, we don't have oc command so we have the freedom to use whatever we want. 

Also, when considering an API call (REST), you'll most likely get an empty object instead of resource not found as you are not searching for a resource here (not specific one)

Reply:
@RazTamir the reason for raising an exception is that if pod_obj does not have spec.volumes[0].persistentVolumeClaim, the failure will be of ```None type doesn't have...``` so we'd like to catch such a thing and to raise a proper exception

Reply:
if we'll return None in case pvc is None it will affect the usage of this function (ocs_ci.ocs.resources.pod.get_pvc_name). If we won't raise an exception here, this test will falsely pass

Reply:
This comment was supposed to be in line 1043 where we explicitly raised the exception.

Look at this https://www.restapitutorial.com/httpstatuscodes.html.
The implementation here is similar to 404 but that's not right as we are not asking to GET for specific resource. 

Your call, not blocking this PR

Reply:
Approving the PR

Reply:
@RazTamir what is the better option then?

Reply:
Returning the pvc name (str) or None if doesn't exist

Reply:
let's continue the discussion here https://github.com/red-hat-storage/ocs-ci/issues/1946.
Merging this one